x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -216,11 +217,13 @@ public CsvTests(String fileName, String groupName, String testName, Integer line
 
     public final void test() throws Throwable {
         try {
+            assumeTrue("Test " + testName + " is not enabled", isEnabled(testName, Version.CURRENT));
+
             /*
-             * We're intentionally not NodeFeatures here because we expect all
-             * of the features to be supported in this unit test.
+             * The csv tests support all but a few features. The unsupported features
+             * are tested in integration tests.
              */
-            assumeTrue("Test " + testName + " is not enabled", isEnabled(testName, Version.CURRENT));
+            assumeFalse("metadata fields aren't supported", testCase.requiredFeatures.contains(EsqlFeatures.METADATA_FIELDS.id()));
```

Comment:
Yeah! Sorry, that's a sad side effect of dropping that version thingy - which we're supposed to be doing too. I figured I'd backport to 8.14 and ignore 8.13 because we're not likely to cut any more of those.